### PR TITLE
Add docs for saveOrFail and deleteOrFail

### PR DESCRIPTION
**Pull request by cleptric**
As promised in cakephp/cakephp#10164, here are the docs for `Cake\ORM\Table::saveOrFail()` and `Cake\ORM\Table::deleteOrFail()`.
I hope I covered everything necessary.

On a site note, using the provided Dockerfile worked like a charme 👍

### CODE REVIEW

File: en/orm/saving-data.rst [hunk 1]
```diff
@@ -1106,6 +1106,31 @@ receiving from the end user is the correct type. Failing to correctly handle
...
+Strict Saving
+=============
+
+.. php:method:: saveOrFail($entity, $options = [])
+
+
+Using this method will throw an :php:exc:`Cake\\ORM\\Exception\\PersistenceFailedException`
+if the application rules checks failed, the entity contains errors or the save was aborted by a callback.
+Using this can be helpful when you performing complex database operations without human monitoring,
+for example, inside a Shell task.
```
Comment (dereuromark) [inline en/orm/saving-data.rst]: Should we add:

Do not use this for validation inside controller actions and where feedback to the user has to be given in forms.

etc?

Comment (cleptric) [inline en/orm/saving-data.rst]: Yes, good point!

**Comment by inoas** [comment]
Now you made me feel all fuzzy about the docker file. Is it available in general on some repo or who is the sourc? @cleptric  ;)

File: en/orm/saving-data.rst (hunk 1 above)
Comment (markstory) [inline en/orm/saving-data.rst]: One could use it there if they catch the exception.

**Comment by antograssiot** [comment]
@inoas you should have a look to the readme.

File: en/orm/saving-data.rst (hunk 1 above)
Comment (dereuromark) [inline en/orm/saving-data.rst]: Yeah but not without catching it then :)

Comment (markstory) [inline en/orm/saving-data.rst]: 'Do not use' is pretty strongly worded when it _can_ be used if you catch the resulting exceptions.

File: en/orm/deleting-data.rst [hunk 2]
```diff
@@ -76,3 +76,26 @@ A bulk-delete will be considered successful if 1 or more rows are deleted.
...
+If you want to track down the entity that failed to save, you can use the
+:php:meth:`Cake\\ORM\Exception\\PersistenceFailedException::getEntity()` method::
+
+        try {
+            $table->deleteOrFail($entity);
+        } catch (\Cake\ORM\Exception\PersistenceFailedException $e) {
+            echo $e->getEntity();
+        }
+
+As this internally perfoms a :php:meth:`Cake\\ORM\\Table::delete()` call, all corresponding delete events will be triggered.
```
Comment (bcrowe) [inline en/orm/deleting-data.rst]: s/perfoms/performs

Comment (markstory) [inline en/orm/deleting-data.rst]: I'll fix that post-merge.

**Pull request merged**